libraries/AP_HAL_SITL/sitl_compass.cpp

```diff
@@ -26,21 +26,14 @@ using namespace HALSITL;
   setup the compass with new input
   all inputs are in degrees
  */
-void SITL_State::_update_compass(float rollDeg, float pitchDeg, float yawDeg)
+void SITL_State::_update_compass(void)
 {
     static uint32_t last_update;
 
     if (_compass == nullptr) {
         // no compass in this sketch
         return;
     }
-    yawDeg += _sitl->mag_error;
```

Comment:
Does this mean SIM_MAG_ERROR parameter doesn't do anything? We should remove it then.